Update readme to human readable (#39)
crossplane/addon-oam-kubernetes-local: README.md

The main target audience of this repository is OAM workloads/traits/scopes developers. If you are looking for end user experience of OAM application on Crossplane, you probably want to check its [getting started doc](https://crossplane.io/docs/v0.11/getting-started/install-configure.html) instead.

## Prerequisites

1. go version 1.12+
2. Kubernetes version v1.16+ with kubectl configured
3. Helm 3.0+ 
4. [Crossplane](https://github.com/crossplane/crossplane) is installed as OAM Kubernetes runtime

## (Optional) In case you don't know how to install Crossplane

Here's a minimal version:
```console
kubectl create namespace crossplane-system
helm repo add crossplane-alpha https://charts.crossplane.io/alpha
helm install crossplane --namespace crossplane-system crossplane-alpha/crossplane
```

Please feel free to ignore this warning, which is cause by helm v2 compatibility issue: 
```
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
```

## Install OAM controllers

#### Clone repo

```console
git clone git@github.com:crossplane/addon-oam-kubernetes-local.git
cd ./addon-oam-kubernetes-local
```

> #### Install cert manager

Author: Not sure if we still need this part.

Reviewer: I think we have to remove webhook along with cert manager if we want to change K8s version requirement to 1.16

Author: webhook is already disabled, this part can be optional.

Reviewer 2: I tested on minikube and it works with K9s version 1.16.2+, it does not work with 1.16.0 or 1.16.1 due to errors on Crossplane installation on those versions.

# OAM Controllers

This repo contains Kubernetes controllers for [core](https://github.com/oam-dev/spec/blob/master/3.workload.md#core-workload) and [standard](https://github.com/oam-dev/spec/blob/master/3.workload.md#standard-workload) OAM workloads/traits/scopes. 

> **👷 WARNNING:** This repo is designed to be used by **OAM workloads/traits/scopes developers.**

> > If you are looking for end user experience of OAM application on Crossplane, you probably want to check its

Reviewer 3: ```suggestion
>  We would suggest you check  [**getting started guide&rarr;**](https://crossplane.io/docs/v0.11/getting-started/install-configure.html) if you are looking for an end-user experience of running OAM applications on the Crossplane.
```

> [**getting started &rarr;**](https://crossplane.io/docs/v0.11/getting-started/install-configure.html)

## Prerequisites

- Kubernetes v1.16+
- Helm 3
- [Crossplane](https://github.com/crossplane/crossplane) v0.11+ installed

## (Optional) In case you don't know how to install Crossplane

> Here's a minimal version:

Reviewer 3: ```suggestion
Here's an easy three steps version:
```

```console
kubectl create namespace crossplane-system
helm repo add crossplane-alpha https://charts.crossplane.io/alpha
helm install crossplane --namespace crossplane-system crossplane-alpha/crossplane
```

> Please feel free to ignore this warning, which is cause by helm v2 compatibility issue: 

Reviewer 3: ```suggestion
Please feel free to ignore this warning which is caused by helm v2 compatibility issue: 
```

```
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
```

## Install OAM controllers

#### Clone this repo

```console
git clone git@github.com:crossplane/addon-oam-kubernetes-local.git
cd ./addon-oam-kubernetes-local
```

#### (Optional) Enable webhook and install cert manager

> In OAM `ManualScalerTrait` controller, it includes a webhook to [validate its workload reference](https://github.com/crossplane/addon-oam-kubernetes-local/blob/757d1922a5266e775b1f131af7da4fb6cbc1a037/pkg/webhooks/manualscalertrait_webhook.go). This webhook [is disabled by default](https://github.com/crossplane/addon-oam-kubernetes-local/blob/42e82c49fb679df4e295802b4727c25faaad4d24/charts/oam-core-resources/values.yaml#L6) in OAM controllers.

Reviewer 3: ```suggestion
The OAM `ManualScalerTrait` controller includes a sample webhook to [validate the manual scalar trait](https://github.com/crossplane/addon-oam-kubernetes-local/blob/757d1922a5266e775b1f131af7da4fb6cbc1a037/pkg/webhooks/manualscalertrait_webhook.go). This webhook [is disabled by default](https://github.com/crossplane/addon-oam-kubernetes-local/blob/42e82c49fb679df4e295802b4727c25faaad4d24/charts/oam-core-resources/values.yaml#L6) in its helm chart.
```

> If you want to enable it, the cert manager will be needed to secure the webhook server:

Reviewer 3: ```suggestion
You need to install a cert-manager to provide self-signed certifications if you want to play with the webhooks.
```

```console
kubectl create namespace cert-manager
kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.14.0/cert-manager.yaml
```
> For more detailed instructions of cert manager please check [Cert-manager docs](https://cert-manager.io/docs/installation/kubernetes/).

#### Install controllers

```console
kubectl create namespace oam-system
helm install controller -n oam-system ./charts/oam-core-resources/ 
```

## Verify

* Apply a sample application configuration

```console
kubectl apply -f examples/containerized-workload/
```

* Verify that the application is running
You should see a deployment looking like below
```console
kubectl get deployments
NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
example-appconfig-workload-deployment   10/10   10           10          8m11s
```

And a service looking like below
```console
kubectl get services
AME                                            TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE

> example-appconfig-workload-deployment-service   ClusterIP   10.96.78.215   <none>        8080/TCP   8m28s

Reviewer 3: The default serviceType  is NodeType now
